src/main/java/com/thealgorithms/conversions/UnitsConverter.java

```diff
@@ -0,0 +1,88 @@
+package com.thealgorithms.conversions;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.lang3.tuple.Pair;
+
+public final class UnitsConverter {
+    private final Map<Pair<String, String>, AffineConverter> conversions;
```

Comment:
For me it is better to use `Pair<K, V>`, it is short and clear.
But if you want to achieve simplicity, what do you think to simplify Pair usage and introduce such class, for easier reading:

```
public final class ConversionKey {
    private final String inputUnit;
    private final String outputUnit;

    public ConversionKey(String inputUnit, String outputUnit) {
        this.inputUnit = inputUnit;
        this.outputUnit = outputUnit;
    }

    public String getInputUnit() {
        return inputUnit;
    }

    public String getOutputUnit() {
        return outputUnit;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        ConversionKey that = (ConversionKey) o;
        return Objects.equals(inputUnit, that.inputUnit) &&
               Objects.equals(outputUnit, that.outputUnit);
    }

    @Override
    public int hashCode() {
        return Objects.hash(inputUnit, outputUnit);
    }

    @Override
    public String toString() {
        return "ConversionKey{" +
                "inputUnit='" + inputUnit + '\'' +
                ", outputUnit='" + outputUnit + '\'' +
                '}';
    }
}
```

Reply:
All of the classes like `AffineConverter`, `UnitsConverter` are not really meant to be used by anyone.